src/Chapter2.hs

```diff
@@ -857,7 +889,8 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+rewind :: [a] -> [a]
+rewind list = map (list !!) [length list - 1, length list - 2..0]
```

Comment:
Creative solution! 🎨 

Though it is slow as `!!`, `length` walks each element of the list to get its size, so on each step, you will go through all the elements, which will take a lot! And imagine making both of them for each element at every step.
Also `!!` is not a safe function and can fail in runtime if the function will be outside the length of the list.

That is why it is always better to rewrite it with the `:` cons. Remember the explanation with trains? 🚂 🚋 🚋 

That is why a more efficient solution is with the accumulator and the recursive function that will do the addition at the start of the list which is instant!

You can read a bit more about the `go` pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#recursive-go

Reply:
Thanks for the suggestions! I took some more time for this one, and might have cheated a bit but I understand it a lot better now :D